integrations/clip/clip_onnx_export.py

```diff
@@ -0,0 +1,229 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Examples on how to use sparsemls's onnx export functionality to export CLIP visual
+and text models using the OpenCLIP API.
+
+Note: This requires torch nightly and openclip to be installed:
+https://github.com/mlfoundations/open_clip
+
+"""
+import argparse
+from collections import OrderedDict
+from pathlib import Path
+from typing import Any, Union
+
+import torch
+
+import open_clip
+from clip_models import TextModel, VisualModel
+from sparseml.pytorch.utils import export_onnx
+
+
+def _export_onnx(
+    module: torch.nn.Module,
+    sample_batch: Any,
+    file_path: Union[Path, str],
+    opset: int = 14,
+    **export_kwargs,
+):
+    export_onnx(
+        module=module,
+        sample_batch=sample_batch,
+        opset=opset,
+        file_path=file_path,
+        **export_kwargs,
+    )
+
+
+def _export_visual(
+    model: torch.nn.Module,
+    device: str,
+    export_path: Union[str, Path],
+    is_coca: bool,
+    **export_kwargs,
+):
+    module_name = "clip_visual.onnx"
+    visual_model = VisualModel(
+        visual_model=model.visual,
+        output_tokens=is_coca,
+    )
+
+    image_shape = visual_model.visual_model.image_size[0]
+    sample_input = torch.randn(1, 3, image_shape, image_shape, requires_grad=True)
+
+    visual_model = visual_model.to(device)
+    visual_model.eval()
+
+    _export_onnx(
+        module=visual_model,
+        sample_batch=sample_input,
+        file_path=export_path / module_name,
+        **export_kwargs,
+    )
+
+
+def _export_text(
+    model: torch.nn.Module,
+    device: str,
+    export_path: Union[str, Path],
+    tokenizer,
+    is_coca: bool,
+    **export_kwargs,
+):
+    module_name = "clip_text.onnx"
```

Comment:
Would this fail or overwrite the model if existing `clip_text.onnx` is present? Same comment for other hardcoded names

Reply:
It won't fail. Will just write over the existing files.